src/main/java/com/frameworkium/core/ui/tests/BaseTest.java

```diff
@@ -1,29 +1,40 @@
 package com.frameworkium.core.ui.tests;
 
-import com.frameworkium.core.common.listeners.*;
+import com.frameworkium.core.common.listeners.MethodInterceptor;
+import com.frameworkium.core.common.listeners.ResultLoggerListener;
+import com.frameworkium.core.common.listeners.TestListener;
 import com.frameworkium.core.common.reporting.TestIdUtils;
 import com.frameworkium.core.common.reporting.allure.AllureLogger;
 import com.frameworkium.core.common.reporting.allure.AllureProperties;
 import com.frameworkium.core.ui.capture.ScreenshotCapture;
-import com.frameworkium.core.ui.driver.*;
-import com.frameworkium.core.ui.listeners.*;
+import com.frameworkium.core.ui.driver.Driver;
+import com.frameworkium.core.ui.driver.DriverSetup;
+import com.frameworkium.core.ui.driver.WebDriverWrapper;
+import com.frameworkium.core.ui.listeners.CaptureListener;
+import com.frameworkium.core.ui.listeners.SauceLabsListener;
+import com.frameworkium.core.ui.listeners.ScreenshotListener;
 import com.saucelabs.common.SauceOnDemandAuthentication;
 import com.saucelabs.common.SauceOnDemandSessionIdProvider;
 import com.saucelabs.testng.SauceOnDemandAuthenticationProvider;
 import org.apache.commons.lang.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.openqa.selenium.NoSuchElementException;
-import org.openqa.selenium.*;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.remote.SessionId;
 import org.openqa.selenium.support.ui.FluentWait;
 import org.openqa.selenium.support.ui.Wait;
 import org.testng.annotations.*;
 
 import java.lang.reflect.Method;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Optional;
-import java.util.concurrent.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
```

Comment:
We should agree a format. I have IntelliJ to automatically optimise imports and it loves doing this when I go in a class.  I'd say it's best practise to import what you actually need, rather than them all. For example, selenium libraries were wild-carded even though it only used 2.